File: in_Java/0807 Max Increase to Keep City Skyline.java
class Solution {
    public int maxIncreaseKeepingSkyline(int[][] grid) {
        int[] rowview = new int[grid.length];
        int[] colview = new int[grid[0].length];
        int cur;
        for(int i = 0; i < grid.length; i++){
            cur = Integer.MIN_VALUE;
            for(int j = 0; j < grid[0].length; j++){
                if(cur < grid[i][j]) cur = grid[i][j];
            }
            rowview[i] = cur;
        }
        for(int i = 0; i < grid[0].length; i++){
            cur = Integer.MIN_VALUE;
            for(int j = 0; j < grid.length; j++){
                if(cur < grid[j][i]) cur = grid[j][i];
            }
            colview[i] = cur;
        }
        int res = 0;
        for(int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[0].length; j++){
                res += Math.min(rowview[i],colview[j]) - grid[i][j];
            }
        }
        return res;
    }
}

File: in_Python/0807 Max Increase to Keep City Skyline.py
class Solution(object):
    def maxIncreaseKeepingSkyline(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        res = 0
        row = len(grid)
        col = len(grid[0])
        rowtop = []
        coltop = []
        for i in range(row):
            cur = 0
            for j in range(col):
                if grid[i][j] > cur:
                    cur = grid[i][j]
            rowtop.append(cur)
        for i in range(col):
            cur = 0
            for j in range(row):
                if grid[j][i] > cur:
                    cur = grid[j][i]
            coltop.append(cur)
        for i in range(row):
            for j in range(col):
                res += min(rowtop[i], coltop[j]) - grid[i][j]
        return res